src/zodbpickle/_pickle_27.c

```diff
@@ -6170,6 +6221,10 @@ init_stuff(PyObject *module_dict)
         return -1;
     if (PyType_Ready(&Picklertype) < 0)
         return -1;
+    BinaryType_t.tp_basicsize = PyString_Type.tp_basicsize;
```

Comment:
I think copying the sizes could be omitted. But I wasn't 100% sure, and since matching the sizes is a big part of this I went explicit.